File: Avocado.Web/Models/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace Avocado.Web.Models
{
    public class RegisterModel : LoginModel
    {
        [Required]
        public string Name { get; set; }
        [Required]
        public string UserName { get; set; }
        public string Picture { get; set; }
    }
}

File: Avocado.Web/ClientApp/src/app/auth/models/RegisterModel.ts
import { LoginModel } from "@avocado/auth/models/LoginModel";

export class RegisterModel extends LoginModel {
  picture: string;
  constructor(
    public name: string,
    public userName: string,
    public provider: string,
    public accessToken: string
  ) {
    super(provider, accessToken);
  }
}
